### PR TITLE
Implement blocks with results (non-zero arity)

### PR DESCRIPTION
**Pull request by chfast**
Requires #25, #67.

### CODE REVIEW

File: lib/fizzy/execute.cpp [hunk 1]
```diff
@@ -271,12 +264,16 @@ execution_result execute(Instance& instance, FuncIdx function, std::vector<uint6
...
             immediates = label.immediate;
 
-            // FIXME: handle arity
-            assert(label.arity == 0);
-
             // When branch is taken, additional stack items must be dropped.
-            assert(stack.size() >= label.stack_height);
-            stack.resize(label.stack_height);
+            assert(stack.size() >= label.stack_height + label.arity);
+            if (label.arity)
```
Comment (gumb0) [inline lib/fizzy/execute.cpp]:
```suggestion
            if (label.arity > 0)
```

Comment (axic) [reply lib/fizzy/execute.cpp]: Why not `!= 0` instead?

Comment (chfast) [reply lib/fizzy/execute.cpp]: Probably the most correct one is `== 1`. Maybe I will just add an assert inside.

File: lib/fizzy/execute.cpp [hunk 2]
```diff
@@ -271,12 +264,17 @@ execution_result execute(Instance& instance, FuncIdx function, std::vector<uint6
...
-            // FIXME: handle arity
-            assert(label.arity == 0);
-
             // When branch is taken, additional stack items must be dropped.
-            assert(stack.size() >= label.stack_height);
-            stack.resize(label.stack_height);
+            assert(stack.size() >= label.stack_height + label.arity);
+            if (label.arity != 0)
+            {
+                assert(label.arity == 1);
```
Comment (chfast) [inline lib/fizzy/execute.cpp]: Fine now?

**Pull request merged**